common/models/templates/src/tokenization.ts

```diff
@@ -33,6 +33,42 @@ namespace models {
     let leftSpans  = wordBreaker(context.left || '') || [];
     let rightSpans = wordBreaker(context.right || '') || [];
 
+    let leftTail: Span;
+    if(leftSpans.length > 0) {
+      leftTail = leftSpans[leftSpans.length - 1];
+    }
+
+    // Handle any directly-pre-caret 'detokenization' cases - where for this _specific_ context,
+    // we should not make a token division where one normally would exist otherwise.
+    // Refer to https://github.com/keymanapp/keyman/issues/6572.
+    if(leftSpans.length > 1 && rightSpans.length == 0) {
+      const leftTailBase = leftSpans[leftSpans.length - 2];
+
+      // if the two spans are adjacent...
+      // Nested if statements in case we need another 'detokenization' condition in the future;
+      // the outer condition should hold for all such cases.
+      if(leftTailBase.end == leftTail.start) {
+        // Ideal:  if(leftTailBase is standard-char-class && leftTail is single-quote-class)
+        // But we don't have character class access here; it's all wordbreaker-function internal.
+        // Upon inspection of the wordbreaker data definitions... the single-quote-class is ONLY "'".
+        // So... we'll just be lazy for now and append the `'`.
+        if(leftTail.text == "'") {
```

Comment:
Will we need to use a new field for lexical-models to declare what character they use for contractions? (in languages that have contractions).

Or just limit to apostrophes for now?

Reply:
+1 on this question.

Reply:
Perhaps, if they overload punctuation marks in the same manner that English does.  Refer to https://unicode.org/reports/tr29/#WB_Rule_Macros.  That said, I just checked against https://en.wikipedia.org/wiki/Contraction_(grammar) - I don't see examples involving punctuation aside from `'`.

Note the class `Single_Quote` on the Unicode wordbreaker spec page - its usage in rules are responsible for the effects we're seeing.

While reading that page, I also remembered that in some languages `'` (more likely, a similar-looking code point) is an actual _letter_ in the language, so there's a chance an issue might arise there.

Reply:
Also, I remember that at one point, @eddieantonio had suggested facilitating character class overrides or something of the sort to allow a degree of language-specific configurability for the word-breaker.  That never happened, though.  

I think his suggestion was motivated by issues we noted with its insufficiency for handling Lao and Khmer, as words for those languages break after nearly every character by default, if I recall correctly.

Reply:
> While reading that page, I also remembered that in some languages ' (more likely, a similar-looking code point) is an actual letter in the language, so there's a chance an issue might arise there.

Apostrophe has been commonly used (but shouldn't be) for a number of languages, such as Hawaiian, for a glottal stop. The alternative preferred character is [U+02BB](https://www.fileformat.info/info/unicode/char/02bb/index.htm). A model for Hawaiian should manage that conversion. But that shouldn't impact this PR I think?

Reply:
> I think his suggestion was motivated by issues we noted with its insufficiency for handling Lao and Khmer, as words for those languages break after nearly every character by default, if I recall correctly.

I think that the best approach is custom wordbreakers for those languages, rather than trying to tweak the default one or making the default one adjustable.